pyuvm/s24_pyuvm_reg_includes.py

```diff
@@ -1,121 +1,178 @@
 '''
 Collection of defines to be sued
 '''
+
 from enum import Enum
-from tabnanny import check
-#import vsc
-import copy
-from pyuvm.error_classes import UVMFatalError, UVMError
-from pyuvm.s17_pyuvm_reg_enumerations import *
+from pyuvm.error_classes import UVMFatalError, UVMError, UVMNotImplemented
+
 
+# path_t
 class path_t(Enum):
     '''
     Access TYPE
     '''
-    FRONTDOOR       = 1
-    BACKDOOR        = 2
-    USER_FRONTDOOR  = 3
-    
+    FRONTDOOR = 1
+    BACKDOOR = 2
+    USER_FRONTDOOR = 3
+
+
+# check_t
 class check_t(Enum):
     '''
-    Check TYPE    
+    Check TYPE
     '''
-    CHECK       = 1
-    NO_CHECK    = 2
+    CHECK = 1
+    NO_CHECK = 2
 
-class elem_kind(Enum):
+
+# status_t
+class status_t(Enum):
     '''
     Status TYPE
     '''
-    IS_OK       = 1
-    IS_NOT_OK   = 2
+    IS_OK = 1
+    IS_NOT_OK = 2
+
 
-## Predict Type
+# Predict Type
 class predict_t(Enum):
     '''
     predict_t main prediction to be used
     PREDICT_WRITE   = 1
     PREDICT_READ    = 2
-    PREDICT_DIRECT  = 3    
-    '''
-    PREDICT_WRITE   = 1
-    PREDICT_READ    = 2
     PREDICT_DIRECT  = 3
+    '''
+    PREDICT_WRITE = 1
+    PREDICT_READ = 2
+    PREDICT_DIRECT = 3
 
-##
+
+#
 class elem_kind_e(Enum):
     pass
 
-##
+
+#
 class access_e(Enum):
     '''
     access_e typoe of access allowed
     PYUVM_READ  = 0
-    PYUVM_WRITE = 1    
-    '''
-    PYUVM_READ  = 0
     PYUVM_WRITE = 1
+    '''
+    UVM_READ = 0
+    UVM_WRITE = 1
+
 
-class pyuvm_resp_t(Enum):
+#
+class uvm_resp_t(Enum):
     '''
-    pyuvm_resp_t is the main response based on the access issued
+    uvm_resp_t is the main response based on the access issued
     PASS_RESP  = 0
     ERROR_RESP = 1
     '''
-    PASS_RESP  = 0
+    PASS_RESP = 0
     ERROR_RESP = 1
 
+
+#
 def rand_enable(use_pyvsc: bool):
     '''
     New Decorator class with randomization option
     If the randomization is switched off then the decorator will no more use
     py_vsc but it just disables it allowing user to use local methods if needed
     '''
     class enable_rand:
-        ## Accept class as argument
+        # Accept class as argument
         def __init__(self, cls) -> None:
             self.cls = cls
 
-        ## operate on CLS init inupt argument    
+        # operate on CLS init inupt argument
         def __call__(self):
-            if(use_pyvsc == True):
-                pass
-                #return vsc.randobj
-            else: 
-                # Return the function unchanged, not decorated. if use_pyvsc is not enabled
+            if (use_pyvsc is True):
+                raise UVMNotImplemented()
+            else:
+                # Return the function unchanged, not decorated. if use_pyvsc
+                # is not enabled
                 return self.cls
     return enable_rand
 
+
```

Comment:
up to here:

- replaced `PYUVM` with `UVM` in enum values
- used raise UVMNotImplemented() in case of use_pyvsc is set to 1
- fixed all the flake8 errors